sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -343,10 +343,19 @@ bool handleError(pi_result Error, const device_impl &DeviceImpl,
   case PI_INVALID_VALUE:
     return handleInvalidValue(DeviceImpl, NDRDesc);
 
+  case PI_PLUGIN_SPECIFIC_ERROR: {
+    char *message = nullptr;
+    DeviceImpl.getPlugin().call_nocheck<PiApiKind::piPluginGetLastError>(
```

Comment:
if this call returns PI_SUCCESS, then this is a warning, not error, and we should not throw, I think

Reply:
The current behaviour of `handleError` is to throw an error for all Error values. If a warning is issued then this function should not be called.

Reply:
I think @smaslov-intel has a great point. It has not been applicable to kernel launch (at least not that I have seen) thus far, but there is no reason we couldn't have a warning from `SetKernelParamsAndLaunch` in which case there we shouldn't throw an exception. You are right that this was not the semantics of `handleError` before, but luckily we have the power to change that. 😉 

What I suggest is to change the name from `handleError` to `handleErrorOrWarning` and make it return `void` instead of bool (because we know it will either throw or be a warning, so bool really doesn't mean much.) That way, in case of a warning it will just fall down to returning `PI_SUCCESS`. In this case we should however reintroduce the warning message... Maybe it would make sense to just call `checkPiResult` here like in the previous iteration but break instead of throw after the call?

Reply:
Calling `checkPiResult` means that the exception will be handled by the call to `__SYCL_CHECK_OCL_CODE_THROW` within that function call. So any mechanisms to print the error message for `PI_PLUGIN_SPECIFIC_ERROR` would need to be implemented there in `sycl/include/CL/sycl/detail/common.hpp`. Do you think this is a better approach than the current one?

Reply:
Or perhaps `checkPiResult` could be modified so that it doesn't call `__SYCL_CHECK_OCL_CODE_THROW`

Reply:
Another option is to add another argument to `__SYCL_CHECK_OCL_CODE_THROW` (and then `__SYCL_REPORT_OCL_ERR_TO_EXC_THROW` and then `__SYCL_REPORT_OCL_ERR_TO_EXC`) to pass along the additional message that `checkPiResult` read, having it ignore it if it is a `nullptr`.